homeassistant/components/n26/sensor.py

```diff
@@ -0,0 +1,294 @@
+"""Support for N26 bank account sensors.
+
+The sensor allows to show
+* N26 account balance
+* Spaces (sub-accounts within an account) balance
+* bank card status
+"""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.n26 import (
+    ACCOUNT_INFO, API, CARDS, DOMAIN as COMPONENT_DOMAIN, SPACES,
+    timestamp_ms_to_date)
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['n26']
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=10)
+
+ATTR_IBAN = "account"
+ATTR_USABLE_BALANCE = "usable_balance"
+ATTR_AVAILABLE_BALANCE = "available_balance"
+
+ATTR_ACC_OWNER_TITLE = "owner_title"
+ATTR_ACC_OWNER_FIRST_NAME = "owner_first_name"
+ATTR_ACC_OWNER_LAST_NAME = "owner_last_name"
+ATTR_ACC_OWNER_GENDER = "owner_gender"
+ATTR_ACC_OWNER_BIRTH_DATE = "owner_birth_date"
+ATTR_ACC_OWNER_EMAIL = "owner_email"
+ATTR_ACC_OWNER_PHONE_NUMBER = "owner_phone_number"
+
+ICON_ACCOUNT = 'mdi:currency-eur'
+ICON_CARD = 'mdi:credit-card'
+ICON_SPACE = 'mdi:crop-square'
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the N26 sensor platform."""
+    api = hass.data[COMPONENT_DOMAIN][API]
+    account_info = hass.data[COMPONENT_DOMAIN][ACCOUNT_INFO]
+    balance = api.get_balance()
+    limits = api.get_account_limits()
+    account_statuses = api.get_account_statuses()
+
+    sensor_entities = [N26Account(
+        api, account_info, account_statuses, limits, balance)]
+
+    cards = hass.data[COMPONENT_DOMAIN][CARDS]
+    for card in cards:
+        sensor_entities.append(N26Card(api, balance["iban"], card))
```

Comment:
What if balance is `None` or doesn't have `iban`

Reply:
Well that would be pretty strange :smile: but I can add a check for that.